### PR TITLE
Option to not have caret go past EOL

### PR DESCRIPTION
**Pull request by WarPie**
GUIs + Slacky = bad idea (specially these auto-genrated GUIs).. but it seems I managed to get it right.
As asked for here: https://villavu.com/forum/showthread.php?t=115984

Settings

### CODE REVIEW

File: Projects/Simba/simbasettingssimple.lfm [hunk 1]
```diff
@@ -140,75 +140,75 @@ object SettingsSimpleForm: TSettingsSimpleForm
...
           Top = 50
-          Width = 149
+          Width = 121
           Caption = 'Check every (minutes):'
           ParentColor = False
         end
         object UpdateMinutesEdit: TEdit
           Left = 160
-          Height = 25
+          Height = 23
```
Comment (nielsAD) [inline Projects/Simba/simbasettingssimple.lfm]: Weird that the height of other TEdits and TLabels has changed?

File: Projects/Simba/simbasettingssimple.pas [hunk 2]
```diff
@@ -220,6 +229,16 @@ procedure TSettingsSimpleForm.ButtonOKClick(Sender: TObject);
   end;
 end;
 
+procedure TSettingsSimpleForm.CodeToolsCheckBoxesClick(Sender: TObject);
+begin
```
Comment (nielsAD) [inline Projects/Simba/simbasettingssimple.pas]: Unnecessary?

File: Projects/Simba/simbasettingssimple.pas [hunk 3]
```diff
@@ -302,6 +322,26 @@ procedure TSettingsSimpleForm.FormShow(Sender: TObject);
   SettingsTabState[2] := 0;
 end;
 
+procedure TSettingsSimpleForm.GroupBox2Click(Sender: TObject);
+begin
```
Comment (nielsAD) [inline Projects/Simba/simbasettingssimple.pas]: More unnecessary auto-generated (?) function bodies?

**Comment by nielsAD** [comment]
Looks good apart from the empty functions. They are automatically generated when you double click on a component in the form editor, so I guess that happened a few times?

File: Projects/Simba/simbasettingssimple.lfm (hunk 1 above)
Comment (WarPie) [inline Projects/Simba/simbasettingssimple.lfm]: Magic

File: Projects/Simba/framescript.pas [hunk 4]
```diff
@@ -767,10 +767,14 @@ constructor TScriptFrame.Create(TheOwner: TComponent);
   OwnerSheet.Caption:= ScriptName;
 
   SynEdit.Enabled := True; // For some reason we need this?
-
+
```
Comment (MerlijnWajer) [inline Projects/Simba/framescript.pas]: Two spaces? ;)

**Comment by MerlijnWajer** [comment]
Is there a reason we have the width changes? Or shall we remove them?

**Comment by WarPie** [comment]
I'd assume it's because the `.lfm` files where for Lazarus `1.0` where I use a more updated version: Laz1.4

**Comment by nielsAD** [comment]
Thanks, @WarPie !

**Pull request merged**